applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp

```diff
@@ -1268,35 +1269,35 @@ void UPwSmallStrainElement<TDim, TNumNodes>::CalculateAndAddCouplingTerms(Vector
 {
     KRATOS_TRY
 
-    rVariables.UPMatrix = (-1.0) * GeoTransportEquationUtilities::CalculateCouplingMatrix(
-                                       rVariables.B, this->GetStressStatePolicy().GetVoigtVector(),
-                                       rVariables.Np, rVariables.BiotCoefficient,
-                                       rVariables.BishopCoefficient, rVariables.IntegrationCoefficient);
-    rVariables.UVector = prod(rVariables.UPMatrix, rVariables.PressureVector);
-    GeoElementUtilities::AssembleUBlockVector(rRightHandSideVector, rVariables.UVector);
+    BoundedMatrix<double, TNumNodes * TDim, TNumNodes> coupling_matrix =
+        (-1.0) * GeoTransportEquationUtilities::CalculateCouplingMatrix(
+                     rVariables.B, this->GetStressStatePolicy().GetVoigtVector(), rVariables.Np,
+                     rVariables.BiotCoefficient, rVariables.BishopCoefficient, rVariables.IntegrationCoefficient);
+
+    array_1d<double, TNumNodes * TDim> coupling_terms = prod(coupling_matrix, rVariables.PressureVector);
+    GeoElementUtilities::AssembleUBlockVector(rRightHandSideVector, coupling_terms);
 
     if (!rVariables.IgnoreUndrained) {
         const double SaturationCoefficient = rVariables.DegreeOfSaturation / rVariables.BishopCoefficient;
-        rVariables.PVector = PORE_PRESSURE_SIGN_FACTOR * SaturationCoefficient *
-                             prod(trans(rVariables.UPMatrix), rVariables.VelocityVector);
-        GeoElementUtilities::AssemblePBlockVector(rRightHandSideVector, rVariables.PVector);
+        array_1d<double, TNumNodes> p_vector = PORE_PRESSURE_SIGN_FACTOR * SaturationCoefficient *
+                                               prod(trans(coupling_matrix), rVariables.VelocityVector);
+        GeoElementUtilities::AssemblePBlockVector(rRightHandSideVector, p_vector);
     }
 
     KRATOS_CATCH("")
 }
 
 template <unsigned int TDim, unsigned int TNumNodes>
-void UPwSmallStrainElement<TDim, TNumNodes>::CalculateCompressibilityFlow(
-    BoundedMatrix<double, TNumNodes, TNumNodes>& rPMatrix,
-    array_1d<double, TNumNodes>&                 rPVector,
-    const ElementVariables&                      rVariables) const
+array_1d<double, TNumNodes> UPwSmallStrainElement<TDim, TNumNodes>::CalculateCompressibilityFlow(const ElementVariables& rVariables) const
 {
     KRATOS_TRY
 
-    noalias(rPMatrix) = GeoTransportEquationUtilities::CalculateCompressibilityMatrix(
-        rVariables.Np, rVariables.BiotModulusInverse, rVariables.IntegrationCoefficient);
+    BoundedMatrix<double, TNumNodes, TNumNodes> p_matrix =
+        GeoTransportEquationUtilities::CalculateCompressibilityMatrix(
+            rVariables.Np, rVariables.BiotModulusInverse, rVariables.IntegrationCoefficient);
+    array_1d<double, TNumNodes> result = -prod(p_matrix, rVariables.DtPressureVector);
```

Comment:
Can't we combine this statement with the next one?
```suggestion
    return -prod(p_matrix, rVariables.DtPressureVector);
```

Reply:
Thank you. It is done in other functions.